### PR TITLE
[Concurrency] Fix too eager early return in checkIsolation mode detecting

### PR DESCRIPTION
**Pull request by ktoso**
Description: If we have the ability to check env variables, and there was no env variable set we aggressively bailed out of the function's logic, without setting the executor mode at all. This prevented the dynamic modes from working in real (non testing) scenarios.

Thanks to @mikeash for helping spot the silly mistake

Resolves rdar://127400013

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please smoke test

File: stdlib/public/Concurrency/Actor.cpp [hunk 1]
```diff
@@ -351,15 +351,14 @@ static void checkIsCurrentExecutorMode(void *context) {
...
-  
-  if (strcmp(modeStr, "nocrash") == 0) {
-    useLegacyMode = Legacy_NoCheckIsolated_NonCrashing;
-  } else if (strcmp(modeStr, "crash") == 0)  {
-    useLegacyMode = Default_UseCheckIsolated_AllowCrash;
-  } // else, just use the platform detected mode
+  if (const char *modeStr =
+          getenv("SWIFT_IS_CURRENT_EXECUTOR_LEGACY_MODE_OVERRIDE")) {
+    if (strcmp(modeStr, "nocrash") == 0) {
+      useLegacyMode = Legacy_NoCheckIsolated_NonCrashing;
```
Comment (mikeash) [inline stdlib/public/Concurrency/Actor.cpp:357]: Should set true/false here. useLegacyMode is a bool and this ends up relying on the enum cases being 0 and 1.

**Review by mikeash — APPROVED** [review]

File: stdlib/public/Concurrency/Actor.cpp (hunk 1 above)
Comment (ktoso) [reply stdlib/public/Concurrency/Actor.cpp:357]: Will do, thank you

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
This was a pain but made use now of the proper env-var way to declare these env vars -- solution was silly but took a while to find. This is much cleaner now and will show up in help advice. The option exists only for us in testing really, but it's good to document it properly.

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Pull request merged**